src/actions/threads.ts

```diff
@@ -1,31 +1,45 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import {ThreadTypes} from 'action_types';
+import {ThreadTypes, PostTypes} from 'action_types';
 import {Client4} from 'client';
+
 import ThreadConstants from 'constants/threads';
+
 import {DispatchFunc, GetStateFunc} from 'types/actions';
+
+import {UserThreadList} from 'types/threads';
+
 import {logError} from './errors';
 import {forceLogoutIfNecessary} from './helpers';
 
 export function getThreads(userId: string, teamId: string, {page = 0, perPage = ThreadConstants.THREADS_CHUNK_SIZE} = {}) {
     return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
-        let threads;
+        let userThreadList: undefined | UserThreadList;
 
         try {
-            threads = await Client4.getUserThreads(userId, teamId, page, perPage);
+            userThreadList = await Client4.getUserThreads(userId, teamId, {page, pageSize: perPage, extended: true});
         } catch (error) {
             forceLogoutIfNecessary(error, dispatch, getState);
             dispatch(logError(error));
             return {error};
         }
 
+        dispatch({
+            type: PostTypes.RECEIVED_POSTS,
+            data: {posts: userThreadList.threads.map(({post}) => post)},
+        });
+
         dispatch({
             type: ThreadTypes.RECEIVED_THREADS,
-            data: {threads, team_id: teamId},
+            data: {
+                ...userThreadList,
+                threads: userThreadList.threads.map((thread) => ({...thread, is_following: true})),
```

Comment:
`is_following` added to locally track/manage; all `UserThreads` returned here are presumed to be _followed_.